backend/routers/purchase_history.py

```
@@ -0,0 +1,245 @@
+            status_code=500,
+            detail={
+                "status": "error",
+                "code": "purchase/recent-fetch-failed",
+                "message": f"Failed to fetch recent purchase history: {str(e)}"
+            }
+        )
+
+@router.get("/{purchase_id}", response_model=PurchaseHistoryResponse)
+async def get_purchase_history_by_id(
+    purchase_id: str,
+    current_user: User = Depends(get_current_user_mock),
```

Ruff (0.14.4)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)